src/components/App/ExpressRecorder.tsx

```diff
@@ -504,18 +505,16 @@ export class ExpressRecorder extends Component<ExpressRecorderProps, State> {
         toggleChanged?: boolean
     ) => {
         this.modifyConstraints(constraints).then(finalConstraints => {
-            if (!finalConstraints.video && !finalConstraints.audio) {
-                this.setState({
-                    error: this.translator.translate(
-                        "Video and audio are disabled, at least one of them must be enabled."
-                    )
-                });
-                return;
-            }
-            this.setState({
-                constraints: finalConstraints
+            const isMainStreamOn = finalConstraints.video || (finalConstraints.audio && !screenOn);
+            this.setState((prevState: State) => {
+                return {
+                    constraints: finalConstraints,
+                    stream: isMainStreamOn ? prevState.stream : undefined,
```

Comment:
warning with that. once again, you will lose control of execution time.
you might be setting `stream` state from line **512** and line **522**.
if they should run in parallel, then ignore this.